modules/Rock.cmake

```diff
@@ -765,28 +759,85 @@ endfunction()
 # files, they get added to the library and the corresponding header file is
 # passed to moc.
 function(rock_testsuite TARGET_NAME)
+    rock_test_common(${TARGET_NAME} ${ARGN})
+    rock_setup_boost_test()
+    rock_add_test()
+endfunction()
+
+## Uses gtest + google-mock as unit testing framework
+# The interface is the same as the one from rock_testsuite
+#
+# Unfortunately, gtest and google-mock do not ship with
+# pre-compiled libraries so we have to add their source files
+# to the test target. If GTEST_DIR or GMOCK_DIR are not set,
+# this function will look for the required source files
+# in /usr/src, which is the default path under ubuntu/debian
+function(rock_gtest TARGET_NAME)
+    if (NOT GTEST_DIR)
+        find_path(GTEST_DIR "src/gtest-all.cc" /usr/src/gtest)
+    endif()
+    if (NOT GMOCK_DIR)
+        find_path(GMOCK_DIR "src/gmock-all.cc" /usr/src/gmock)
+    endif()
+
+    if (NOT GTEST_DIR)
+        message(FATAL_ERROR "Could not find gtest in /usr/src")
+    endif()
+    if (NOT GMOCK_DIR)
+        message(FATAL_ERROR "Could not find google-mock in /usr/src")
+    endif()
+    message(STATUS "gtest found ... building the test suite")
+
+    rock_test_common(${TARGET_NAME} ${GTEST_DIR}/src/gtest-all.cc
+                                    ${GMOCK_DIR}/src/gmock-all.cc
+                                    ${ARGN})
+
+    rock_setup_gtest_test()
+    rock_add_test()
+endfunction()
+
+macro(rock_setup_gtest_test)
+    target_include_directories(${TARGET_NAME} SYSTEM PUBLIC ${GMOCK_DIR} ${GTEST_DIR}
+                               ${GMOCK_DIR}/include ${GTEST_DIR}/include)
+    target_link_libraries(${TARGET_NAME} pthread)
+
+    if (ROCK_TEST_LOG_DIR)
+        list(APPEND __rock_test_parameters
+             --gtest_output=xml:${ROCK_TEST_LOG_DIR}/${TARGET_NAME}.gtest.xml)
+        file(MAKE_DIRECTORY "${ROCK_TEST_LOG_DIR}")
+    endif()
+endmacro()
+
+function(rock_test_common TARGET_NAME)
     if (TARGET_NAME STREQUAL "test")
         message(WARNING "test name cannot be 'test', renaming to '${PROJECT_NAME}-test'")
         set(TARGET_NAME "${PROJECT_NAME}-test")
     endif()
+
+    rock_executable(${TARGET_NAME} ${ARGN} NOINSTALL)
+endfunction()
+
+macro(rock_setup_boost_test)
```

Comment:
Can we append __rock_test_parameters in one function and use it in a different one?